test/Mono.Linker.Tests.Cases/DataFlow/GenericParameterDataFlow.cs

```diff
@@ -766,9 +766,7 @@ static void TestNoWarningsInRUCMethod<T> ()
 
 		// Warn about calls to the static methods and the ctor on the RUC type:
 		[ExpectedWarning ("IL2026", "RUCTypeRequiresPublicFields<T>.StaticMethod", "message")]
-		[ExpectedWarning ("IL2026", "RUCTypeRequiresPublicFields<T>.StaticMethodRequiresPublicMethods<U>", "message", ProducedBy = ProducedBy.Trimmer)]
-		// https://github.com/dotnet/linker/issues/2573
-		[ExpectedWarning ("IL2026", "RUCTypeRequiresPublicFields<T>.StaticMethodRequiresPublicMethods<T>", "message", ProducedBy = ProducedBy.Analyzer)]
+		[ExpectedWarning ("IL2026", "RUCTypeRequiresPublicFields<T>.StaticMethodRequiresPublicMethods<U>", "message")]
```

Comment:
Here the generic type declaration happens to have the same name as the instantiation because both use `T` for the generic parameter name. Would you mind adding another test where the type is instantiated over something other than `T`?